modular_skyrat/modules/contractor/code/items/baton.dm

```diff
@@ -52,24 +69,36 @@
 
 /obj/item/melee/baton/telescopic/contractor_baton/additional_effects_non_cyborg(mob/living/carbon/target, mob/living/user)
 	. = ..()
-	if(!((upgrade_flags & BATON_MUTE_UPGRADE) && !(upgrade_flags & BATON_ALL_UPGRADE))|| !istype(target))
+	if(!istype(target))
 		return
-	if(target.silent >= (MUTE_CYCLES * 2))
-		return
-	target.silent += MUTE_CYCLES //10 seconds of mute a hit up to 20 max
+	if((upgrade_flags & BATON_MUTE_UPGRADE) || (upgrade_flags & BATON_ALL_UPGRADE))
+		if(target.silent < (MUTE_CYCLES * MUTE_MAX_MOD))
+			target.silent += MUTE_CYCLES //10 seconds of mute a hit up to 20 max
+	if((upgrade_flags & BATON_FOCUS_UPGRADE) || (upgrade_flags & BATON_ALL_UPGRADE))
+		if(target == user.mind?.opposing_force?.contractor_hub?.current_contract?.contract.target?.current) // Pain
+			target.apply_damage(BONUS_STAMINA_DAM, STAMINA, BODY_ZONE_CHEST)
+			target.stuttering += BONUS_STUTTER
 
 /obj/item/melee/baton/telescopic/contractor_baton/upgraded
 	upgrade_flags = BATON_ALL_UPGRADE
 	desc = "A compact, specialised baton assigned to Syndicate contractors. Applies light electrical shocks to targets. This one seems to be fully upgraded with unremovable parts."
 
+/obj/item/melee/baton/telescopic/contractor_baton/upgraded/Initialize(mapload)
+	. = ..()
+	for(var/num in 1 to CUFF_MAXIMUM)
+		new/obj/item/restraints/handcuffs/cable(src)
+
+/obj/item/melee/baton/telescopic/contractor_baton/upgraded/wrench_act(mob/living/user, obj/item/tool)
+	return
+
 /obj/item/baton_upgrade
 	icon = 'modular_skyrat/modules/contractor/icons/baton_upgrades.dmi'
 	var/upgrade_flag
 
 /obj/item/baton_upgrade/proc/desc_update(obj/item/melee/baton/telescopic/contractor_baton/batong)
 	if(!batong.upgrade_flags)
-		desc += "<br><br>[span_boldnotice("[batong] has the following upgrades attached:")]"
-	desc += "<br>[span_notice("[src].")]"
+		batong.desc += "<br><br>[span_boldnotice("[batong] has the following upgrades attached:")]"
+	batong.desc += "<br>[span_notice("[src].")]"
```

Comment:
That can probably go in the baton's `examine()` proc instead.

Reply:
Done